### PR TITLE
Feat 2622 s2s truststore setup

### PR DESCRIPTION
**Pull request by vlad-ciucescu**

### CODE REVIEW

File: sormas-base/setup/import-to-truststore.sh [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+      echo "sormas2sormas.truststorePass=${SORMAS_S2S_TRUSTSTORE_PASS}";
+    } >> "${SORMAS_PROPERTIES}"
+fi
+else
+  # export existing certificates to temporary file
+  TEMP_FILE=${SORMAS2SORMAS_DIR}/tempcert.pem
+  openssl pkcs12 -in "${TRUSTSTORE_FILE}" -password pass:"${SORMAS_S2S_TRUSTSTORE_PASS}" -out ${TEMP_FILE}
+
+  # create new truststore with the new certificate and the certificates from the temporary file
+  openssl pkcs12 -export -nokeys -out "${TRUSTSTORE_FILE}" -password pass:"${SORMAS_S2S_TRUSTSTORE_PASS}" -in "${CRT_FILE}" -certfile ${TEMP_FILE}
```
Comment (HolgerReiseVSys) [inline sormas-base/setup/import-to-truststore.sh:100]: It seems that the java Keystore ignores plain certificates without alias/friendlyName.
With openssl there seems to be no way to add several certificates with friendlyNames.
So we probably should require Java to be installed and use keytool instead.

File: sormas-base/setup/import-to-truststore.sh [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+fi
+
+# DIRECTORIES
+if [[ ${LINUX} = true ]]; then
+	ROOT_PREFIX=
+else
+	ROOT_PREFIX=/c
+fi
+if [[ -z "${SORMAS2SORMAS_DIR}" ]] || [[ ! -d "${SORMAS2SORMAS_DIR}" ]]; then
+  DEFAULT_SORMAS2SORMAS_DIR="${ROOT_PREFIX}/opt/sormas2sormas"
```
Comment (HolgerReiseVSys) [inline sormas-base/setup/import-to-truststore.sh:39]: If a SORMAS2SORMAS_DIR is specified, but does not exist, the script should fail.

File: sormas-base/setup/generate-cert.sh [hunk 3]
```diff
@@ -69,30 +70,39 @@ echo "The certificate will be generated with the following subject:"
...
-PEM_NAME=${SORMAS2SORMAS_DIR}/sormas2sormas.privkey.pem
-P12_NAME=${SORMAS2SORMAS_DIR}/sormas2sormas.keystore.p12
-CRT_NAME=${SORMAS2SORMAS_DIR}/sormas2sormas.cert.crt
+PEM_FILE=${SORMAS2SORMAS_DIR}/sormas2sormas.privkey.pem
+P12_FILE=${SORMAS2SORMAS_DIR}/sormas2sormas.keystore.p12
+CRT_FILE=${SORMAS2SORMAS_DIR}/sormas2sormas.cert.crt
 
 # generate private key and self signed certificate
-openssl req -sha256 -newkey rsa:4096 -passout pass:"${SORMAS_S2S_CERT_PASS}" -keyout "${PEM_NAME}" -x509 -passin pass:"${SORMAS_S2S_CERT_PASS}" -days 1095 -subj "${CERT_SUBJ}" -out "${CRT_NAME}"
+openssl req -sha256 -newkey rsa:4096 -passout pass:"${SORMAS_S2S_CERT_PASS}" -keyout "${PEM_FILE}" -x509 -passin pass:"${SORMAS_S2S_CERT_PASS}" -days 1095 -subj "${CERT_SUBJ}" -out "${CRT_FILE}"
```
Comment (HolgerReiseVSys) [inline sormas-base/setup/generate-cert.sh:78]: There seems to be a miminum length of 4 characters for the password. Maybe we should check that.

File: sormas-base/setup/import-to-truststore.sh [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#*******************************************************************************
+
+#!/bin/bash
+
+echo "# SORMAS TO SORMAS CERTIFICATE IMPORT"
+echo "# This script imports a certificate into the local truststore, to be used for SORMAS2SORMAS and SurvNet communication"
```
[the remark this answers is not in the document]
Comment (vlad-ciucescu) [reply sormas-base/setup/import-to-truststore.sh:23]: Yeah, removed it for the other ticket to, probably conflict resolution error

**Comment by JonasCir** [comment]
Thrilled to see S2S work soon :D Maybe add a `docs/` root folder and start to place documentation/guides there? The repo base is already pretty crowded :smiley:

**Review by HolgerReiseVSys — CHANGES_REQUESTED** [review]
The scripts should have at least soemthing like `s2s` in their names so it is clearer what they are used for.

File: GUIDE_SORMAS2SORMAS_CERTIFICATE.md [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+4. After providing the requested data, the certificate files will be generated. <br/>
+   The generated certificate has a validity of 3 years. 
+   The certificate files will be available in the root SORMAS directory, in the folder ``/sormas2sormas``.
+5. The generated ``.p12`` file should not be shared with third parties. <br/>
+   The generated ``.crt`` file will be verified and shared with other SORMAS instances, from which this instance
+   will be able to request data. Conversely, in order to enable other SORMAS instances to request data from this 
+   instance, their certificate files should be obtained and added to the local truststore. More details can be found
+   in the next section.
+6. If the ``SORMAS_PROPERTIES`` environment variable is available, the relevant properties will be 
+    automatically set by the script.
```
[the remark this answers is not in the document]
Comment (vlad-ciucescu) [reply GUIDE_SORMAS2SORMAS_CERTIFICATE.md:37]: But the script is for generating a new certificate. I don't think it should matter if another one is available and the properties are already set.

[the remark this answers is not in the document]
Comment (vlad-ciucescu) [reply GUIDE_SORMAS2SORMAS_CERTIFICATE.md:37]: But I can add this functionality in the case of the truststore password

**Review by HolgerReiseVSys — APPROVED** [review]

**Pull request merged**